client/external-declarations.d.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import '@wordpress/date';
```

Comment:
Doing this avoids overriding the whole module with just the exported function below.
If removed, we would only see `datei18n` from this package.